# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1241486556 14400
#      Mon May 04 21:22:36 2009 -0400
# Node ID 7ea76c8a59f75b928a0112321f975ee98d2a650a
# Parent  5baa4d43a2ae715cf51f386b99c3756ce12d3711
configure.in: put -lftgl ahead of $FT2_LIBS, and $FT2_LIBS ahead of $OPENGL_LIBS

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+2009-05-04  John W. Eaton  <jwe@octave.org>
+
+	* configure.in: Put -lftgl ahead of $FT2_LIBS, and $FT2_LIBS ahead
+	of $OPENGL_LIBS.
+
 2009-05-04  Peter O'Gorman  <pogma@thewrittenword.com>
 
 	* aclocal.m4 (OCTAVE_HAVE_C99_VSNPRINTF): New macro.
 	* configure.in: Use it.
 
 	* configure.in (rs6000-ibm-aix* | powerpc-ibm-aix*): Set
 	library_path_var to LIBPATH.
 	(hppa*-hp-hpux*): Set library_path_var to SHLIB_PATH.
diff --git a/configure.in b/configure.in
--- a/configure.in
+++ b/configure.in
@@ -693,28 +693,28 @@ if test -n "$OPENGL_LIBS"; then
 ## ftgl (needs freetype 2)
   AC_CHECK_FT2([9.0.3],[],
     [warn_freetype="FreeType library not found. Native renderer will not have on-screen text"])
   if test -z "$warn_freetype"; then
     AC_LANG_PUSH(C++)
       save_LIBS="$LIBS"
       save_CXXFLAGS="$CXXFLAGS"
       save_CPPFLAGS="$CPPFLAGS"
-      LIBS="$LIBS $OPENGL_LIBS $FT2_LIBS"
+      LIBS="$LIBS $FT2_LIBS $OPENGL_LIBS"
       CXXFLAGS="$CXXFLAGS $FT2_CFLAGS"
       CPPFLAGS="$CPPFLAGS $FT2_CFLAGS"
       has_ftgl_h=yes
       AC_CHECK_HEADERS([FTGL/ftgl.h ftgl.h], break, [
 	AC_CHECK_HEADERS([FTGL/FTGL.h FTGL.h], [
-          AC_DEFINE(HAVE_FTGL_UPPERCASE,1,[Define to 1 if you have FTGL.h or FTGL/FTGL.h])
+          AC_DEFINE(HAVE_FTGL_UPPERCASE, 1, [Define to 1 if you have FTGL.h or FTGL/FTGL.h])
 	  break
 	  ], has_ftgl_h=no)])
       if test "$has_ftgl_h" = yes; then
 	AC_MSG_CHECKING([for FTGLTextureFont in -lftgl])
-	LIBS="$LIBS -lftgl"
+	LIBS="$save_LIBS -lftgl $FT2_LIBS $OPENGL_LIBS"
         AC_TRY_LINK([
 #ifdef HAVE_FTGL_FTGL_H
 #ifdef HAVE_FTGL_UPPERCASE
 #include <FTGL/FTGL.h>
 #else
 #include <FTGL/ftgl.h>
 #endif
 #include <FTGL/FTGLTextureFont.h>
@@ -722,22 +722,22 @@ if test -n "$OPENGL_LIBS"; then
 #ifdef HAVE_FTGL_UPPERCASE
 #include <FTGL.h>
 #else
 #include <ftgl.h>
 #endif
 #include <FTGLTextureFont.h>
 #endif], [
 FTGLTextureFont font("");], [
-          OPENGL_LIBS="$OPENGL_LIBS $FT2_LIBS -lftgl"
+          OPENGL_LIBS="-lftgl $FT2_LIBS $OPENGL_LIBS"
 	  LIBS="$save_LIBS"
 	  CPPFLAGS="$save_CPPFLAGS"
           AC_DEFINE(HAVE_FTGL, 1, [Define to 1 if FTGL is present])
 	  AC_MSG_RESULT(yes)
-	  XTRA_CXXFLAGS="$XTRA_CXXFLAGS $FT2_CFLAGS"],[
+	  XTRA_CXXFLAGS="$XTRA_CXXFLAGS $FT2_CFLAGS"], [
 	  AC_MSG_RESULT(no)
           warn_ftgl="FTGL library not found. Native renderer will not have on-screen text"
         ])
 	LIBS="$save_LIBS"
 	CXXFLAGS="$save_CXXFLAGS"
 	CPPFLAGS="$save_CPPFLAGS"
       else
         warn_ftgl="FTGL headers not found. Native renderer will not have on-screen text"
